code/userprog/exception.cc

```diff
@@ -155,7 +160,20 @@ int Release_Syscall(int lockIndex)
 // Deletes a lock from the lock table using an interger argument, IF the lock is not in use. If the lock is in use, it is eventually deleted when the lock is no longer in use.
 int DestroyLock_Syscall(int lockIndex)
 {
-	//TODO
+   /*
+    * check LockIndex numerically for in bounds
+	if (lockIndex <0 || lockIndex >= TABLE_SIZE){
+		char errorMessage[100] = "DestroyLock::Error: Lock Index out of bounds";
+		Write_Syscall(errorMessage,strlen(errorMessage),  ConsoleOutput);
+	}
+	KernelLock* kl = LockTable[lockIndex];
+	if (kl) {
+		if (kl->lockQueueEmpty) {
+		  delete kl->lock;
```

Comment:
We want to follow the same rule we use with KernelCondition deletion, KernelLock needs a toBeDeleted bool that is set to false when DestroyLock is called. then, upon Release, check this bool, and if it's true, delete it there